internal/cloudinfo/machine_types.go

```diff
@@ -85,37 +107,40 @@ func fetchMachineTypes(cloud string, service string, region string, location str
 	var vmDetails CloudInfoResponse
 	json.Unmarshal(respBody, &vmDetails)
 
+	instanceTypeMap.lock.Lock()
```

Comment:
IMHO we never should use `instanceTypeMap.lock` from outside of the enclosing struct.

I'd add a `update(*InstanceTypeMap)` method to `InstanceTypeMap` or `setVM(VMKey, *MachineDetails)` and lock/unlock inside the method